ui/src/tool-panel/location-track/split-store.ts

```diff
@@ -200,14 +251,42 @@ export const splitReducers = {
         if (state.splittingState) {
             if (payload.type === 'SPLIT') {
                 state.splittingState.splits = state.splittingState.splits
-                    .filter((split) => split.switchId !== payload.switchId)
+                    .filter((split) => split.switch.switchId !== payload.switch.switchId)
                     .concat([payload])
                     .sort((a, b) => a.distance - b.distance);
             } else {
                 state.splittingState.firstSplit = payload;
             }
         }
     },
+    setHighlightedSplit: (
+        state: TrackLayoutState,
+        { payload: split }: PayloadAction<SplitTargetId | undefined>,
+    ): void => {
+        if (state.splittingState) {
+            state.splittingState.highlightedSplit = split;
+        }
+    },
+    setFocusedSplit: (
+        state: TrackLayoutState,
+        { payload: split }: PayloadAction<SplitTargetId | undefined>,
+    ): void => {
+        if (state.splittingState) {
+            state.splittingState.focusedSplit = split;
+        }
+    },
+    setHighlightedSwitch: (
+        state: TrackLayoutState,
+        { payload: switchId }: PayloadAction<LayoutSwitchId | undefined>,
+    ): void => {
+        if (state.splittingState) {
+            state.splittingState.highlightedSwitch = switchId;
+            console.log(
+                ' state.splittingState.highlightedSwitch',
+                state.splittingState.highlightedSwitch,
+            );
+        }
+    },
```

Comment:
Hieman ikävää että tällaisia pitää tuoda reduxiin, kun sitten ne jäävät elämään esim. refreshien yli. Oletettavasti ei löytynyt mitään järkevää tapaa tehdä näitä esim. stateiksi jonnekin alemmas? 🤔 

Toisaalta jos joskus saataisiin aikaiseksi se pelkästään muistinvaraisesti reduxissa elävien asioiden store, niin nämä voisi ihan hyvin elää silloin sielläkin.